docs/design.md

```diff
@@ -0,0 +1,104 @@
+# Design and architecture
+
+## Design
+The `image-rs` crate is a rustified and tailored version of [containers/image](https://github.com/containers/image),
+to provide a small, simple, secure, lightweight and high performance OCI container image management library for
+[Confidential Containers](https://github.com/confidential-containers).
+
+Currently the `image-rs` crate focuses on usage scenarios to run `Confidential Containers` on Kubernetes clusters with
+[Kata Containers](https://katacontainers.io/). It may also be enhanced to support other confidential container runtimes
```

Comment:
I'm unclear as to whether "confidential container" here is a general term or the name of this project. If the latter, it should consistently be referred to using leading caps ("Confidential Container").

Reply:
yes, will use leading caps